File: timemanager/app/api/preferences.py
from fastapi import APIRouter, Depends
from ..models.preferences import UserPreferences
from ..db.memory_db import db
from .auth import oauth2_scheme
from .schedules import get_current_user_id

router = APIRouter()

@router.put("", response_model=UserPreferences)
async def update_preferences(
    preferences: UserPreferences,
    current_user_id: str = Depends(get_current_user_id)
):
    preferences_dict = preferences.dict()
    preferences_dict["user_id"] = current_user_id
    updated = db.set_preferences(current_user_id, preferences_dict)
    return UserPreferences(**updated)

@router.get("", response_model=UserPreferences)
async def get_preferences(current_user_id: str = Depends(get_current_user_id)):
    prefs = db.get_preferences(current_user_id)
    if not prefs:
        # Return default preferences
        prefs = {
            "user_id": current_user_id,
            "timezone": "UTC",
            "language": "en",  # Default language is English
            "notification_preferences": {
                "email": True,
                "push": True,
                "wechat": True
            },
            "working_hours": {
                "monday": {"start": "09:00", "end": "17:00"},
                "tuesday": {"start": "09:00", "end": "17:00"},
                "wednesday": {"start": "09:00", "end": "17:00"},
                "thursday": {"start": "09:00", "end": "17:00"},
                "friday": {"start": "09:00", "end": "17:00"}
            },
            "calendar_sync": {
                "google": False,
                "outlook": False,
                "apple": False
            }
        }
        db.set_preferences(current_user_id, prefs)
    return UserPreferences(**prefs)

File: timemanager-windows/src/TimeManager.Client/Models/Preferences.cs
using System;
using System.Collections.Generic;
using CommunityToolkit.Mvvm.ComponentModel;

namespace TimeManager.Client.Models
{
    public partial class WorkingHours : ObservableObject
    {
        [ObservableProperty]
        private string start = "09:00";

        [ObservableProperty]
        private string end = "17:00";
    }

    public partial class NotificationPreferences : ObservableObject
    {
        [ObservableProperty]
        private bool email = true;

        [ObservableProperty]
        private bool push = true;

        [ObservableProperty]
        private bool wechat = true;
    }

    public partial class CalendarSync : ObservableObject
    {
        [ObservableProperty]
        private bool google = false;

        [ObservableProperty]
        private bool outlook = false;

        [ObservableProperty]
        private bool apple = false;
    }

    public partial class UserPreferences : ObservableObject
    {
        [ObservableProperty]
        private string? userId;

        [ObservableProperty]
        private string timezone = "UTC";

        [ObservableProperty]
        private string language = "en";

        [ObservableProperty]
        private NotificationPreferences notificationPreferences = new();

        [ObservableProperty]
        private Dictionary<string, WorkingHours> workingHours = new()
        {
            { "monday", new WorkingHours() },
            { "tuesday", new WorkingHours() },
            { "wednesday", new WorkingHours() },
            { "thursday", new WorkingHours() },
            { "friday", new WorkingHours() }
        };

        [ObservableProperty]
        private CalendarSync calendarSync = new();
    }
}
